src/solver/optimisation/adequacy_patch_local_matching/adequacy_patch_weekly_optimization.cpp

```diff
@@ -45,27 +45,33 @@ AdequacyPatchOptimization::AdequacyPatchOptimization(PROBLEME_HEBDO* problemeHeb
 }
 void AdequacyPatchOptimization::solve(uint weekInTheYear, int hourInTheYear)
 {
-    problemeHebdo_->adqPatchParams->AdequacyFirstStep = true;
-    OPT_OptimisationHebdomadaire(problemeHebdo_, thread_number_);
+    // problemeHebdo_->adqPatchParams->AdequacyFirstStep = true;
+    // OPT_OptimisationHebdomadaire(problemeHebdo_, thread_number_);
+
+    // never set AdequacyFirstStep to True and skip the islanding optimization
```

Comment:
You don't need to change the definition of this class (revert all changes).

We made a nice factory method just for that, see file **src/solver/optimisation/base_weekly_optimization.cpp**. Just return `std::make_unique<DefaultWeeklyOptimization>` in all cases (adq patch enabled/disabled), it's much simpler.